### PR TITLE
UI/payments manual entry

### PR DESCRIPTION
**Pull request by Shruti-Apte**
## Notion card
https://www.notion.so/saeloun/Frontend-User-should-be-able-to-add-payments-entry-manually-against-and-invoice-c65e211eeb9748c38e60ee26d9c2ff4d
## Summary

- UI for Manual payment entry added.
## Preview

https://www.loom.com/share/ef9b7c94de804851a471a0ab9ff7d77e
## Type of change

- [ ] New feature (non-breaking change which adds functionality)

### Checklist:

- [ ] I have manually tested all workflows
- [ ] I have performed a self-review of my own code

### CODE REVIEW

File: app/javascript/src/components/payments/Header.tsx [hunk 1]
```diff
@@ -1,22 +1,34 @@
 import React from "react";
-import { MagnifyingGlass } from "phosphor-react";
+import { MagnifyingGlass,Plus } from "phosphor-react";
```
Comment (onkar-saeloun) [inline app/javascript/src/components/payments/Header.tsx:2]: indent properly `import { MagnifyingGlass, Plus } from "phosphor-react";`

File: app/javascript/src/components/payments/Modals/AddManualEntry.tsx [hunk 2]
```diff
@@ -0,0 +1,267 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
+import React, { useState } from "react";
+import Select, { DropdownIndicatorProps, components } from "react-select";
+import CustomDatePicker from "common/CustomDatePicker";
+import dayjs from "dayjs";
+import { X, Calendar } from "phosphor-react";
+import { MagnifyingGlass } from "phosphor-react";
+
+const AddManualEntry = ({ setShowManualEntryModal, Invoiceoptions }) => {
```
Comment (onkar-saeloun) [inline app/javascript/src/components/payments/Modals/AddManualEntry.tsx:9]: `Invoiceoptions ` should be `invoiceOptions ` please follow camel case

File: app/javascript/src/components/payments/Modals/AddManualEntry.tsx [hunk 3]
```diff
@@ -0,0 +1,267 @@
...
+        ref={innerRef}
+        {...innerProps}
+        className=" p-2 flex justify-between items-center"
+      >
+        <div className="pr-6 pl-0 py-2.5 text-left">
+          <h1 className="text-base font-bold leading-5 text-miru-dark-purple-1000">
+            {props.data.label}
+          </h1>
+          <h3 className="pt-1 font-normal text-sm text-miru-dark-purple-400 leading-5">
+            CB680
```
Comment (onkar-saeloun) [inline app/javascript/src/components/payments/Modals/AddManualEntry.tsx:71]: Why this is hard coded?

File: app/javascript/src/components/payments/Modals/AddManualEntry.tsx [hunk 4]
```diff
@@ -0,0 +1,267 @@
...
+                    <option value="Visa">Visa</option>
+                    <option value="Mastercard">Mastercard</option>
+                    <option value="Bank Transfer">Bank Transfer</option>
+                    <option value="ACH">ACH</option>
+                    <option value="Amex">Amex</option>
+                    <option value="Cash">Cash</option>
+                    <option value="Cheque">Cheque</option>
+                    <option value="Credit Card">Credit Card</option>
+                    <option value="Debit">Debit</option>
+                    <option value="PayPal">PayPal</option>
```
Comment (onkar-saeloun) [inline app/javascript/src/components/payments/Modals/AddManualEntry.tsx:201]: Should be rendered with array.map

File: app/javascript/src/components/payments/index.tsx [hunk 5]
```diff
@@ -1,9 +1,15 @@
...
 import Pagination from "common/Pagination";
 import Header from "./Header";
+import AddManualEntry from "./Modals/AddManualEntry";
 import Table from "./Table/index";
 
 const Payments = () => {
+
+  const [showManualEntryModal, setShowManualEntryModal] = React.useState<boolean>(false);
+
+  const Invoiceoptions = [{ value: "invoice1", label: "Invoice1" },{ value: "invoice2", label: "Invoice2" },{ value: "invoice3", label: "Invoice3" }];
```
Comment (onkar-saeloun) [inline app/javascript/src/components/payments/index.tsx:11]: Are we going to show these values on prod?

**Review by onkar-saeloun — COMMENTED** [review]
@Shruti-Apte few comments.

File: app/javascript/src/components/payments/index.tsx (hunk 5 above)
Comment (Shruti-Apte) [reply app/javascript/src/components/payments/index.tsx:11]: API is not ready yet ..This is a dummy data... What should I add in that array?

File: app/javascript/src/components/payments/Modals/AddManualEntry.tsx (hunk 4 above)
Comment (Shruti-Apte) [reply app/javascript/src/components/payments/Modals/AddManualEntry.tsx:201]: okay

File: app/javascript/src/components/payments/Modals/AddManualEntry.tsx (hunk 3 above)
Comment (Shruti-Apte) [reply app/javascript/src/components/payments/Modals/AddManualEntry.tsx:71]: Same reason ...API is not ready !

File: app/javascript/src/components/payments/index.tsx (hunk 5 above)
Comment (onkar-saeloun) [reply app/javascript/src/components/payments/index.tsx:11]: I think you should not show incorrect data. It's better to keep empty option.

**Comment by supriya3105** [comment]
@Shruti-Apte On selecting invoice , the invoice number - client name should be populated. Also the invoice amount should be automatically populated. Is this something that will be handled during integration work on can be handled with this PR ? 

Please see the designs 
image

**Comment by Shruti-Apte** [comment]
> @Shruti-Apte On selecting invoice , the invoice number - client name should be populated. Also the invoice amount should be automatically populated. Is this something that will be handled during integration work on can be handled with this PR ?
> 
> Please see the designs image

Yes...With integration PR. I need data from backend to do that!

**Comment by supriya3105** [comment]
@onkar-saeloun @ajinkyaa Please review this PR . @apoorv1316 Please test

**Review by onkar-saeloun — APPROVED** [review]

**Review by apoorv1316 — APPROVED** [review]

**Pull request merged**